mypy.bzl

```diff
@@ -91,8 +86,12 @@ def _mypy_rule_impl(ctx, is_aspect = False, exe = None, out_path = None):
     transitive_srcs_depsets = []
     stub_files = []
 
-    if hasattr(base_rule.attr, "srcs"):
-        direct_src_files = _extract_srcs(base_rule.attr.srcs)
+    if not hasattr(base_rule.attr, "srcs"):
+        return None
+    direct_src_files = _extract_srcs(base_rule.attr.srcs)
+    direct_src_root_paths = sets.to_list(
```

Comment:
Doing this looks right. My single `--package-root` looks hacky and wrong, but it wasn't breaking things from what I can see. Did you encounter problems with  `--package-root .`?

Why not also transitive `srcs` as `--package-roots` too?

Reply:
Thanks for reviewing.

I encountered the package root issue trying to depend on the Google protobuf python target using @com_google_protobuf//:protobuf_python, same as #11. However, after fixing the source roots, I found another issue: that target includes \_\_init\_\_.py's under [compatibility_test/v2.5.0](https://github.com/protocolbuffers/protobuf/tree/3be9322e28dfcacdb669f646f76fe833fa274107/python/compatibility_tests/v2.5.0/tests), which is not a valid Python module name, and since the mypy invocation includes all transitive sources, mypy fails on the bad module name. I ended up switching to depending on PyPI's protobuf, since that's packaged correctly (I think @com_google_protobuf//:protobuf_python is broken, in this case).

Now, after switching to protobuf from PyPI, I have a new issue: MYPYPATH isn't populated with imports from transitive dependencies. This causes mypy to emit tons of type errors for the protobuf library because mypy sees a different import path for the external dependency (something like `external.pypi__...__protobuf.google.protobuf`), so it fails to match it with [typeshed's definitions](https://github.com/python/typeshed/tree/00f5240b88f382411510de434ec575b4ae46bcee/third_party/2and3/google/protobuf). I haven't figured out how to fix this yet—let me know if you see an obvious solution.

Reply:
Sounds like more of a mypy issue? The default `mypi.ini` is rather aggressive, if you turn off imports using 

```
[mypy]
follow_imports=skip
```

You might see a solution to your problem. A less shotgun approach is to use [`# type: ignore`](https://github.com/python/mypy/issues/500#issuecomment-86340780). I just took a look at my code, which works with #11 and I have:

```python
from google.protobuf import text_format  # type: ignore
```

In my mypy_test files.

Sorry I haven't taken the time to review this. I've been just using the tests not the aspect, so unless you make those changes- I don't have much to input.

But I'd be excited for a canonical solution that supports protobuf :tada:

Reply:
Would this also work as an alternative to `follow_imports=skip`? 

```
[mypy-protobuf.*]
ignore_missing_imports = True
```